src/client/components/Map/Map.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component, createRef } from 'react';
+import { Map, TileLayer, Marker, Popup } from 'react-leaflet';
+import { Icon } from 'leaflet';
+import 'leaflet/dist/leaflet.css';
+import logo from '../../assets/images/treasure.png';
+
+const treasureIcon = new Icon({
+  iconUrl: logo,
+  iconSize: [71, 71],
+});
+
+class Mymap extends Component {
+  state = {
```

Comment:
I suggest breaking this component up into smaller components. Try to think of your program as a tree structure which contains levels of abstractions. Each level should not know anything about the details of the lower levels.

Reply:
😃👍